src/openrct2/paint/tile_element/Paint.Surface.cpp

```diff
@@ -902,6 +902,38 @@ static void viewport_surface_draw_water_side_top(
     viewport_surface_draw_tile_side_top(session, edge, height, terrain, self, neighbour, true);
 }
 
+static std::pair<int32_t, int32_t> surface_get_height_above_water(
+    const TileElement& tileElement, const int32_t height, const int32_t surfaceShape)
+{
+    int32_t local_surfaceShape = surfaceShape;
+    int32_t local_height = height;
+
+    if (tileElement.AsSurface()->GetWaterHeight() > 0)
```

Comment:
Since you are expecting the tile element to be a surface, the parameter into the method should just be `SurfaceTileElement`.